docs/spec/v1beta2/statusupdates.md

```diff
@@ -0,0 +1,138 @@
+# Git Commit Status Updates
```

Comment:
Can't this status update document be part of "Working with Providers" section of Provider spec doc. Most of the content here, like defining each of the spec fields is redundant.

Reply:
Commit status is a different feature to alerting, conflating the two (like we did before) makes the doc really complex, that's reason I've split this in two dedicated docs.

Reply:
But this document now seems to suggest there is another object kind, sparking possible confusion?

Reply:
I don't see where we suggest this is a different kind, it's a different feature. We'll have a new kind as proposed in https://github.com/fluxcd/flux2/pull/2639

Reply:
All other specs outline specific object kinds, and then describe what you can do with them and how you can control their behavior. By introduce another document which does not follow this, but is instead a composition of two objects, this becomes difficult to follow. Especially because of:

> ## Writing a Git commit status provider spec
>
> As with all other Kubernetes config, a Provider needs `apiVersion`, `kind`, and `metadata` fields. The name of an Alert 

which contains conflicting language: "Git commit status provider spec", "Provider" and "Alert". Again, making it hard to follow what precisely is being described.

In addition, by reading just e.g. the `Provider` spec it will be non-apparent to a user that this can be used to issue status updates. Given this, I would say Sunny's suggestion is beneficial and could just contain a description explaining the behavior and an example (like document currently has), not very different from e.g. https://github.com/fluxcd/source-controller/blob/main/docs/spec/v1beta2/buckets.md#excluding-files

Reply:
Ok I've put them back, please see 6842c49

Reply:
Much better, small suggestion to add an additional ref.